DSharpPlus.Commands/Processors/TextCommands/TextCommandProcessor.cs

```diff
@@ -220,6 +174,10 @@ await this.extension.commandErrored.InvokeAsync(this.extension, new CommandError
         await this.extension.CommandExecutor.ExecuteAsync(commandContext);
     }
 
+    /// <inheritdoc cref="TextExtensions.CreateFakeMessageEventArgs(CommandContext, DiscordMessage, string)" />
+    public static MessageCreatedEventArgs CreateFakeMessageEventArgs(CommandContext context, DiscordMessage message, string content) => TextExtensions.CreateFakeMessageEventArgs(context, message, content);

```

Comment:
if possible, i would rather this be a separate file we provide unofficially, rather than providing this API and then potentially pulling it shortly thereafter
cc @DSharpPlus/dsharpplus-developers for more feedback

Reply:
Looking at the previous history of Commands getting big PR's, "shortly pulled" would easily be 6 months to a year. It's worth it.

Reply:
you will however note that Commands getting IoC support will likely entail breaking the processors out into smaller pieces of logic (for extensibility and customizability, and also just for cleanliness) - which we can engineer to solve some use-cases in the near future, considering that's currently all i'm focusing on

Reply:
How is this relevant